prometheus/testutil/testutil_test.go

```diff
@@ -332,27 +332,46 @@ Diff:
 }
 
 func TestScrapeAndCompare(t *testing.T) {
-	const expected = `
+	scenarios := map[string]struct {
+		want        string
+		metricNames []string
+		// expectedErrPrefix if empty, means no fail is expected for the comparison.
+		expectedErrPrefix string
```

Comment:
Why prefix and not exact error? Would be nice to test that missing metric semantics.

Reply:
done